api/filters.go

```diff
@@ -139,14 +139,9 @@ func (api *API) submitFilter(w http.ResponseWriter, r *http.Request) {
 
 	}
 
-	filterOutput := &model.FilterOutput{
-		FilterID:   filter.ID,
-		State:      model.Submitted,
-		Events:     []model.Event{},
-		Dimensions: []model.Dimension{},
-	}
+	filterOutput := api.generateFilterOutput(filter)
```

Comment:
I don't know if I like the idea of creating a function for this, the whole thing could be done in one statement.

Reply:
I'm happy to change that